File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/playback/Callback.java
/**
 *
 * @file
 *
 * @brief Notification callback interface
 *
 * @author vitamin.caig@gmail.com
 *
 */

package app.zxtune.playback;

import app.zxtune.TimeStamp;

public interface Callback {

  /**
   * Called after subscription
   */
  void onInitialState(PlaybackControl.State state);

  /**
   * Called on new state (may be the same)
   */
  void onStateChanged(PlaybackControl.State state, TimeStamp pos);

  /**
   * Called on active item change
   */
  void onItemChanged(Item item);

  /**
   * Called on error happends
   */
  void onError(String e);
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/playback/Callback.kt
/**
 *
 * @file
 *
 * @brief Notification callback interface
 *
 * @author vitamin.caig@gmail.com
 */
package app.zxtune.playback

import app.zxtune.TimeStamp

interface Callback {
    /**
     * Called after subscription
     */
    fun onInitialState(state: PlaybackControl.State)

    /**
     * Called on new state (may be the same)
     */
    fun onStateChanged(state: PlaybackControl.State, pos: TimeStamp)

    /**
     * Called on active item change
     */
    fun onItemChanged(item: Item)

    /**
     * Called on error happends
     */
    fun onError(e: String)
}
